### ISSUE TITLE
Search & filter for User Lists

### ISSUE THREAD

**Issue by Aiz0** [axiel7/AniHyou-android#78]
## Request
Currently the Anime and Manga lists use same search & filtering from the explore tab.

This is a bit awkward and does not match the functionality of the AniList website.

Anime and Manga lists should have their own search & filtering implementation.

This should allow you to search & filter other users lists as well.
## Additional context
If possible having filters persist would be appreciated. The app [AniLib](https://github.com/rev0lgenX/AniLib) does this.

**Comment by axiel7** [comment]
Yes I plan to do this. The current search is just a temporal implementation
